public/data/javascript.json

```diff
@@ -810,6 +810,160 @@
         "author": "0xHouss"
       },
+      {
+        "title": "Memoize Function",
+        "description": "Caches the result of a function based on its arguments to improve performance.",
+        "code": [
+          "const memoize = (func) => {",
+          "  const cache = new Map();",
+          "  return (...args) => {",
+          "    const key = JSON.stringify(args);",
+          "    if (cache.has(key)) {",
+          "      return cache.get(key);",
+          "    }",
+          "    const result = func(...args);",
+          "    cache.set(key, result);",
+          "    return result;",
+          "  };",
+          "};",
+          "",
+          "// Usage:",
+          "const factorial = memoize((n) => (n <= 1 ? 1 : n * factorial(n - 1)));",
+          "console.log(factorial(5)); // Output: 120",
+          "console.log(factorial(5)); // Output: 120 (retrieved from cache)"
+        ],
+        "tags": ["javascript", "memoization", "optimization", "utility"],
+        "author": "axorax"
+      },
+      {
+        "title": "Once Function",
+        "description": "Ensures a function is only called once.",
+        "code": [
+          "const once = (func) => {",
+          "  let called = false;",
+          "  return (...args) => {",
+          "    if (!called) {",
+          "      called = true;",
+          "      return func(...args);",
+          "    }",
+          "  };",
+          "};",
+          "",
+          "// Usage:",
+          "const initialize = once(() => console.log('Initialized!'));",
+          "initialize(); // Output: Initialized!",
+          "initialize(); // No output"
+        ],
+        "tags": ["javascript", "function", "once", "utility"],
+        "author": "axorax"
+      },
+      {
+        "title": "Curry Function",
+        "description": "Transforms a function into its curried form.",
+        "code": [
+          "const curry = (func) => {",
+          "  const curried = (...args) => {",
+          "    if (args.length >= func.length) {",
+          "      return func(...args);",
+          "    }",
+          "    return (...nextArgs) => curried(...args, ...nextArgs);",
+          "  };",
+          "  return curried;",
+          "};",
+          "",
+          "// Usage:",
+          "const add = (a, b, c) => a + b + c;",
+          "const curriedAdd = curry(add);",
+          "console.log(curriedAdd(1)(2)(3)); // Output: 6",
+          "console.log(curriedAdd(1, 2)(3)); // Output: 6"
+        ],
+        "tags": ["javascript", "curry", "function", "utility"],
+        "author": "axorax"
+      },
+      {
+        "title": "Compose Functions",
+        "description": "Composes multiple functions into a single function, where the output of one function becomes the input of the next.",
+        "code": [
+          "const compose = (...funcs) => (initialValue) => {",
+          "  return funcs.reduceRight((acc, func) => func(acc), initialValue);",
```

Comment:
IMO it would make more sens using a normal `reduce`, it makes the function more intuitive to use